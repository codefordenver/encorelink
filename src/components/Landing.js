import React from 'react';
import RegisterContainer from '../containers/RegisterContainer';
import howOrganizer from '../static-assets/img/how/how_organizer.png';
import howMusician from '../static-assets/img/how/how_musician.png';
import howPerformance from '../static-assets/img/how/how_performance.png';

function Landing() {
  return (
    <div className="landing-page">
      <div className="theater">
        <div className="row">
          <div className="small-12 medium-6 large-6 large-offset-1 column align-middle text-center">
            <h1>Match. Play. Contribute.</h1>
            <p>
              Encore Link is the best way to match volunteer musicians with organizations
              who are in need of musicians who donate their time.
            </p>
          </div>
          <div className="small-12 medium-6 large-4 large-offset-1 column align-middle">
            <RegisterContainer />
          </div>
        </div>
      </div>
      <div className="row content-block text-center">
        <div className="small-12 column">
          <h1 className="blue-txt">How It Works</h1>
        </div>
        <div className="column small-12 large-4">
          <h1>1</h1>
          <img src={howOrganizer} alt="Organizers Create Volunteering Spots" />
        </div>
        <div className="column small-12 large-4">
          <h1>2</h1>
          <img src={howMusician} alt="Musicians Search and Schedule Volunteering Spots" />
        </div>
        <div className="column small-12 large-4">
          <h1>3</h1>
          <img src={howPerformance} alt="Musicians Play for Audience" />
        </div>
      </div>
    </div>
  );
}

export default Landing;
