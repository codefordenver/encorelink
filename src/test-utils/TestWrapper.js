import PropTypes from 'prop-types';
import React from 'react';
import { Provider } from 'react-redux';

import configureStore from '../store/configureStore';

const store = configureStore();

const routerKeys = [
  'listenBefore',
  'listen',
  'transitionTo',
  'push',
  'replace',
  'go',
  'goBack',
  'goForward',
  'createKey',
  'createPath',
  'createHref',
  'createLocation',
  'setState',
  'registerTransitionHook',
  'unregisterTransitionHook',
  'pushState',
  'replaceState',
  // '__v2_compatible__',
  'setRouteLeaveHook',
  'isActive'
];

const mockRouterObj = routerKeys.reduce((mockRouter, key) => {
  mockRouter[key] = jest.fn(); // eslint-disable-line no-param-reassign
  return mockRouter;
}, {});

class MockRouter extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired
  };

  static childContextTypes = {
    router: PropTypes.object.isRequired
  };

  getChildContext() {
    return {
      router: mockRouterObj
    };
  }

  render() {
    return this.props.children;
  }
}

const TestWrapper = ({ children }) => {
  return (
    <Provider store={store}>
      <MockRouter>
        { children }
      </MockRouter>
    </Provider>
  );
};

TestWrapper.propTypes = {
  children: PropTypes.node.isRequired
};

export default TestWrapper;
