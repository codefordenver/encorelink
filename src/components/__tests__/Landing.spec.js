import React from 'react';
import renderer from 'react-test-renderer';
import TestWrapper from '../../test-utils/TestWrapper';

import Landing from '../Landing';

test('Landing renders correctly', () => {
  const tree = renderer
    .create(<TestWrapper><Landing /></TestWrapper>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
