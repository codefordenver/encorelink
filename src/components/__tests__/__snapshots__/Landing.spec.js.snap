// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders correctly 1`] = `
<div
  className="landing-page"
>
  <div
    className="theater"
  >
    <div
      className="row"
    >
      <div
        className="small-12 medium-6 large-6 large-offset-1 column align-middle text-center"
      >
        <h1>
          Match. Play. Contribute.
        </h1>
        <p>
          Encore Link is the best way to match volunteer musicians with organizations who are in need of musicians who donate their time.
        </p>
      </div>
      <div
        className="small-12 medium-6 large-4 large-offset-1 column align-middle"
      >
        <div
          className="register"
        >
          <ul
            className="tabs"
          >
            <li
              className="tabs-title is-active"
            >
              <a
                onClick={[Function]}
                style={Object {}}
              >
                Sign up as a Musician
              </a>
            </li>
            <li
              className="tabs-title not-active"
            >
              <a
                onClick={[Function]}
                style={Object {}}
              >
                Sign up as an Organizer
              </a>
            </li>
          </ul>
          <div
            className="tabs-content"
          >
            <div
              className="tabs-panel is-active"
            >
              <form
                className="form-register"
                onSubmit={[Function]}
              >
                <label>
                  Email
                  <input
                    autoFocus={true}
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="text"
                  />
                </label>
                <label>
                  Password
                  <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />
                </label>
                <label
                  className="volunteer-check"
                >
                  <input
                    checked={true}
                    name="isMusician"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </label>
                <label
                  className="terms"
                >
                  <input
                    checked={false}
                    className="terms-check"
                    name="agreeTerms"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  I have read and agree to the
                   
                  <a
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    style={Object {}}
                    tabIndex={0}
                  >
                    terms
                  </a>
                  .
                </label>
                <button
                  className="button secondary"
                  disabled={true}
                  type="submit"
                >
                  Register
                </button>
              </form>
            </div>
            <div
              className="tabs-panel not-active"
            >
              <form
                className="form-register"
                onSubmit={[Function]}
              >
                <label>
                  Email
                  <input
                    autoFocus={true}
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="text"
                  />
                </label>
                <label>
                  Password
                  <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />
                </label>
                <label
                  className="volunteer-check"
                >
                  <input
                    checked={false}
                    name="isMusician"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </label>
                <label
                  className="terms"
                >
                  <input
                    checked={false}
                    className="terms-check"
                    name="agreeTerms"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  I have read and agree to the
                   
                  <a
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    style={Object {}}
                    tabIndex={0}
                  >
                    terms
                  </a>
                  .
                </label>
                <button
                  className="button secondary"
                  disabled={true}
                  type="submit"
                >
                  Register
                </button>
              </form>
            </div>
          </div>
          <p>
            Already have an account?Â 
            <a
              href={undefined}
              onClick={[Function]}
              style={Object {}}
            >
              Log in
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row content-block text-center"
  >
    <div
      className="small-12 column"
    >
      <h1
        className="blue-txt"
      >
        How It Works
      </h1>
    </div>
    <div
      className="column small-12 large-4"
    >
      <h1>
        1
      </h1>
      <img
        alt="Organizers Create Volunteering Spots"
        src="how_organizer.png"
      />
    </div>
    <div
      className="column small-12 large-4"
    >
      <h1>
        2
      </h1>
      <img
        alt="Musicians Search and Schedule Volunteering Spots"
        src="how_musician.png"
      />
    </div>
    <div
      className="column small-12 large-4"
    >
      <h1>
        3
      </h1>
      <img
        alt="Musicians Play for Audience"
        src="how_performance.png"
      />
    </div>
  </div>
</div>
`;
